Mathlib/NumberTheory/Fermat.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2024 Moritz Firsching. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Firsching
+-/
+import Mathlib.Algebra.BigOperators.Group.Finset
+import Mathlib.Algebra.Order.Ring.Basic
+import Mathlib.Algebra.Order.Star.Basic
+import Mathlib.Data.Nat.Prime.Defs
+import Mathlib.Tactic.Ring.RingNF
+
+/-!
+# Fermat numbers
+
+The Fermat numbers are a sequence of natural numbers defined as $2^{2^n} + 1$, for all natural
+numbers $n$.
+
+## Main theorems
+  `coprime_fermat_fermat`: two distinct fermat numbers are coprime.
+-/
+open Nat
+open scoped BigOperators
+
+/-- The Fermat numbers:
+the $n$-th Fermat number $F_n$ is defined as $2^{2^n} + 1$. -/
+def fermat : ℕ → ℕ := fun n ↦ 2 ^ (2 ^ n) + 1
+
+@[simp]
+theorem fermat_zero : fermat 0 = 3 := by rfl
+
+@[simp]
+theorem fermat_one : fermat 1 = 5 := by rfl
+
+@[simp]
+theorem fermat_two : fermat 2 = 17 := by rfl
+
+theorem strictMono_fermat : StrictMono fermat := by
+  apply strictMono_nat_of_lt_succ
+  simp only [fermat, add_lt_add_iff_right, Nat.pow_succ]
+  exact fun n => (pow_lt_pow_iff_right one_lt_two).mpr (by aesop)
+
+theorem two_lt_fermat {n : ℕ} : 2 < fermat n := by
```

Comment:
Is this a more useful statement than `3 ≤ fermat n`?

Reply:
I think both are equally useful, but having both seems like an overkill. I added this one, because I though `fermat n - 2` might appear in more than one theorem on fermat numbers and this way, one can easily remove the nat subtraction, but of course it would also be easy to do that if three_le_fermat was present.